### PR TITLE
feat(statutory): pass application limits per body in event

### PR DESCRIPTION
**Pull request by LeonVreling**
Backend implementation for [MA-123](https://myaegee.atlassian.net/jira/software/c/projects/MA/issues/MA-123)

### CODE REVIEW

**Comment by serge1peshcoff** [comment]
I have an idea on how this can be implemented in a bit better way:

1) in fetchEvent, fetch all the PaxLimits for all the bodies (probably adding a function like getFromBodies to PaxLimits that will return all of the PaxLimits where event is the passed one and bodies is one of the bodies passed as an argument)
2) in getEventPermissions, set `permissions.apply_from_body` or sth like that as a map of booleans (true if you can apply on behalf of this body, false if not)
2) this permissions.apply_from_body would be returned in /events/:id, so you can take it from there (in data.permissions) and use it on frontend

**Review by serge1peshcoff — APPROVED** [review]
lgtm

File: lib/middlewares.js [hunk 1]
```diff
@@ -142,12 +142,17 @@ exports.fetchEvent = async (req, res, next) => {
     req.approveRequest = approveRequest;
     if (req.approveRequest.body && req.approveRequest.body.success) req.approvePermissions = approveRequest.body.data;
 
+    const limits = await Promise.all(
```
Comment (serge1peshcoff) [inline lib/middlewares.js:145]: this one can be moved to

```
    if (req.user) {
        myApplication = await Application.findOne({
            where: {
                user_id: req.user.id,
                event_id: event.id
            }
        });
    }
```

so it won't fail a request if there's no `req.user`

Comment (serge1peshcoff) [reply lib/middlewares.js:145]: also need to default it to [] if no req.user, so it'd still be an array

**Review by serge1peshcoff — APPROVED** [review]

File: lib/middlewares.js (hunk 1 above)
Comment (LeonVreling) [reply lib/middlewares.js]: Yes updated it

**Pull request merged**